react-components/src/communities/view/EditCommunityDialog.js

```diff
@@ -0,0 +1,267 @@
+import { getMessage } from "../../util/I18NWrapper";
+import ids from "../ids";
+import messages from "../messages";
+import styles from "../styles";
+import withI18N from "../../util/I18NWrapper";
+
+import Button from "@material-ui/core/Button/Button";
+import CircularProgress from "@material-ui/core/CircularProgress/CircularProgress";
+import Dialog from "@material-ui/core/Dialog";
+import DialogActions from "@material-ui/core/DialogActions";
+import DialogContent from "@material-ui/core/DialogContent";
+import DialogTitle from "@material-ui/core/DialogTitle";
+import PropTypes from "prop-types";
+import React, { Component, Fragment } from "react";
+import { withStyles } from "@material-ui/core";
+
+/**
+ * @author aramsey
+ * A simple button that allows users to create communities
+ */
+class EditCommunityDialog extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            saveCommunity: false,
+            deleteCommunity: false,
+            leaveCommunity: false,
+            joinCommunity: false,
+            loading: false,
+        };
+
+        [
+            'onDialogBtnClicked',
+            'closeDialog',
+            'handleDialogConfirmed',
+            'handleCloseEditDialog',
+            'cancelSave',
+        ].forEach((fn) => this[fn] = this[fn].bind(this));
+    }
+
+    onDialogBtnClicked(dialogName) {
+        this.setState({[dialogName]: true})
+    }
+
+    closeDialog(dialogName) {
+        this.setState({[dialogName]: false})
+    }
+
+    handleDialogConfirmed(dialogName) {
+        const {community} = this.props;
+
+        this.setState({loading: true});
+        this.props.presenter[[dialogName]](community, () => {
+            this.closeDialog([dialogName]);
+            this.setState({loading: false});
+            this.handleCloseEditDialog();
+        });
+    }
+
+    handleCloseEditDialog() {
+        this.setState({
+            saveCommunity: false,
+        });
+
+        this.props.onClose();
+    }
+
+    cancelSave() {
+        this.setState({
+            saveCommunity: false,
+        });
+    }
+
+    render() {
+        const {
+            community,
+            isCommunityAdmin,
+            isMember,
+            onClose,
+            open,
+            collaboratorsUtil,
+            presenter,
+            currentUser,
+            onCommunitySaved,
+            classes,
+        } = this.props;
+
+        const {
+            saveCommunity,
+            loading
+        } = this.state;
+
+        const communityName = community !== null ? collaboratorsUtil.getSubjectDisplayName(community) : null;
+
+        return (
+            <Fragment>
+                <Dialog open={open}
+                        fullWidth={true}
+                        maxWidth='lg'
+                        onClose={onClose}
+                        id={ids.EDIT_DLG}>
+                    <DialogTitle>
+                        <DialogHeader community={community}
```

Comment:
I think I addressed everything except this.  I'm having issues where my component's messages are getting clobbered by DEDialogHeader's messages (there's an aria-label for `more`).  So any text I try to put into the header cannot be found because it's not in the `messages` file used by DEDialogHeader. I'm not sure how to fix this yet.